tiledb/sm/query/readers/sparse_global_order_reader.cc

```diff
@@ -1847,28 +1877,33 @@ void SparseGlobalOrderReader<BitmapType>::process_slabs(
     return;
   }
 
-  // Make a list of unique result tiles.
-  std::vector<ResultTile*> result_tiles;
-  {
-    std::unordered_set<ResultTile*> found_tiles;
-    for (auto& rcs : result_cell_slabs) {
-      if (rcs.length_ != 0) {
-        if (found_tiles.count(rcs.tile_) == 0) {
-          found_tiles.emplace(rcs.tile_);
-          result_tiles.emplace_back(rcs.tile_);
-        }
-      }
-    }
-  }
-  std::sort(result_tiles.begin(), result_tiles.end(), result_tile_cmp);
-
   // Read a few attributes a a time.
+  std::vector<ResultTile*> result_tiles =
+      result_tiles_to_load(result_cell_slabs, false);
   std::optional<std::string> last_field_to_overflow{std::nullopt};
   uint64_t buffer_idx{0};
+  optional<std::vector<ResultTile*>> result_tiles_agg_only;
   while (buffer_idx < names.size()) {
+    // Generate a list of filtered result tiles for aggregates only fields.
+    bool agg_only = aggregate_only(names[buffer_idx]);
+    if (agg_only && result_tiles_agg_only == nullopt) {
+      result_tiles_agg_only = result_tiles_to_load(result_cell_slabs, true);
+
+      // If we hit an overflow, it might have changed the tiles we need to load.
```

Comment:
I have tried to understand how this snippet works and why it's needed, but I gave up, can you please explain what's happening here and huw recomputing memory usage fixes this?